src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,309 @@
+            }
+
+            if (appsAndFeaturesEntry.DisplayName.empty())
+            {
+                appsAndFeaturesEntry.DisplayName = displayName;
+            }
+            if (appsAndFeaturesEntry.DisplayVersion.empty())
+            {
+                appsAndFeaturesEntry.DisplayVersion = displayVersion;
+            }
+            if (appsAndFeaturesEntry.Publisher.empty())
+            {
+                appsAndFeaturesEntry.Publisher = publisher;
+            }
+            if (appsAndFeaturesEntry.ProductCode.empty())
+            {
+                appsAndFeaturesEntry.ProductCode = packageId;
+            }
+
+            return appsAndFeaturesEntry;
+        }
+
+        void AppendExeExtension(std::string& value)
+        {
+            if (!HasSuffix(value, ".exe"))
+            {
+                value += ".exe";
+            }
+        }
+
+        bool AddPortableLinksDirToPathRegistry(Manifest::ScopeEnum& scope, Utility::Architecture& arch, bool& isPathModified)
+        {
+            Key key;
+            if (scope == Manifest::ScopeEnum::Machine)
+            {
+                key = Registry::Key::CreateKeyAndOpen(HKEY_LOCAL_MACHINE, Normalize(s_PathSubkey_Machine));
+            }
+            else
+            {
+                key = Registry::Key::CreateKeyAndOpen(HKEY_CURRENT_USER, Normalize(s_PathSubkey_User));
+            }
+
+            std::string portableLinksDir = GetPortableLinksLocation(scope, arch).u8string();
+            AICLI_LOG(CLI, Info, << "Adding to Path environment variable: " << portableLinksDir);
+            std::wstring pathKey = { L"Path" };
+            std::string pathValue = key[pathKey]->GetValue<Value::Type::String>();
+            if (pathValue.find(portableLinksDir) == std::string::npos)
+            {
+                std::string modifiedPathValue = pathValue;
+
+                if (modifiedPathValue.back() != ';')
+                {
+                    modifiedPathValue += ";";
+                }
+
+                modifiedPathValue += portableLinksDir + ";";
+                bool result = key.SetKeyValue(pathKey, ConvertToUTF16(modifiedPathValue), REG_EXPAND_SZ);
+                if (result)
+                {
+                    isPathModified = true;
+                }
+            }
+
+            return true;
+        }
+
+        bool AddPortableEntryToUninstallRegistry(Manifest::ScopeEnum& scope, AppsAndFeaturesEntry& entry)
+        {
+            HKEY root = (scope == Manifest::ScopeEnum::Machine) ? HKEY_LOCAL_MACHINE : HKEY_CURRENT_USER;
+            std::wstring productCode = ConvertToUTF16(entry.ProductCode);
+            std::wstring displayName = ConvertToUTF16(entry.DisplayName);
+            std::wstring displayVersion = ConvertToUTF16(entry.DisplayVersion);
+            std::wstring publisher = ConvertToUTF16(entry.Publisher);
+            std::wstring uninstallString = L"winget uninstall --id " + productCode;

```

Comment:
Things we discussed for making this more robust:
- Add new argument `--product-code` for `uninstall` and use it here
- Add value here like `WinGetInstallerType` set to the installer type string
- Add values here detailing the file information so that `uninstall` can properly remove only the file (and directory) that we created

Also, check on all of the ARP values for anything that makes sense for us to add, like `InstallLocation`.  I would probably put `WinGetPackageIdentifier` and `WinGetSourceIdentifier` as well.  Those will be useful in other places.

Reply:
Perhaps also - 
```
InstallDate - Current Date
URLInfoAbout - PackageUrl <if defined>
HelpLink - PublisherSupportUrl <if defined>
EstimatedSize - can be calculated?
```